desc/magnetic_fields/_core.py

```diff
@@ -921,16 +938,57 @@ def compute_magnetic_field(
             # zip does not terminate early
             transforms = transforms * len(self._fields)
 
-        B = 0
-        for i, (field, g, tr) in enumerate(zip(self._fields, source_grid, transforms)):
-            B += field.compute_magnetic_field(
-                coords, params[i % len(params)], basis, source_grid=g, transforms=tr
-            )
+        if compute_A_or_B == "B":
```

Comment:
could simplify this further into a single loop with `getattr(field, op)(coords...)`